src/actions.jl

```diff
@@ -1,32 +1,90 @@
-abstract type AbstractActionHomomorphism{T} end
+abstract type Action end
+abstract type ByPermutations <: Action end
+abstract type ByLinearTransformation <: Action end
 
-struct ExtensionHomomorphism{T,I<:Integer,V,Op} <: AbstractActionHomomorphism{T}
-    features::V
+abstract type InducedActionHomomorphism{T} end
+
+(hom::InducedActionHomomorphism)(g::GroupElement) = induce(action(hom), hom, g)
```

Comment:
This does not work on Julia v1.0. Maybe we can call `induce(action(hom), hom, g)` directly in line 10 and add this line later when Julia v1.0 is dropped.